### PR TITLE
feat(ci): Add cargo cache to each rust workflow

### PR DESCRIPTION
**Pull request by phated**
This adds cargo caching to each rust workflow. I'm currently testing to see how much improvement this makes.

### CODE REVIEW

**Comment by phated** [comment]
Moving this to draft because it doesn't seem to help currently. It looks like we are rebuilding `barretenberg_wrapper` on every step that needs a build.

File: .github/workflows/rust-test.yml [hunk 1]
```diff
@@ -36,6 +36,24 @@ jobs:
         if: ${{ matrix.os == 'macos-latest' }}
         run: brew install llvm libomp cmake
 
+      - name: Setup noir cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/noir_cache/
+          key: noir_cache
```
Comment (phated) [inline .github/workflows/rust-test.yml:44]: @kevaundray I was wondering if there is anything wrong with us caching the noir_cache in GitHub Actions cache? It seems like it restores very quickly and would probably cut down on our spurious mac failures.

Comment (kevaundray) [reply .github/workflows/rust-test.yml:44]: Yeah we could cache it -- For the future, we could also say that we cache all setup files for backends that have first class support

Comment (phated) [reply .github/workflows/rust-test.yml:50]: Sounds good. I've updated to clean up some logic, so we're good for a review 👍

**Pull request merged**